core/src/test/java/org/elasticsearch/search/aggregations/bucket/children/ParentToChildrenAggregatorTests.java

```diff
@@ -0,0 +1,167 @@
+import org.elasticsearch.index.mapper.MapperService;
+import org.elasticsearch.index.mapper.NumberFieldMapper;
+import org.elasticsearch.index.mapper.ParentFieldMapper;
+import org.elasticsearch.index.mapper.TypeFieldMapper;
+import org.elasticsearch.index.mapper.Uid;
+import org.elasticsearch.index.mapper.UidFieldMapper;
+import org.elasticsearch.search.aggregations.AggregatorTestCase;
+import org.elasticsearch.search.aggregations.metrics.min.InternalMin;
+import org.elasticsearch.search.aggregations.metrics.min.MinAggregationBuilder;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ParentToChildrenAggregatorTests extends AggregatorTestCase {
+
+    private static final String CHILD_TYPE = "child_type";
+    private static final String PARENT_TYPE = "parent_type";
+
+    public void testNoDocs() throws IOException {
+        testCase(new MatchAllDocsQuery(), iw -> {
+            // Intentionally not writing any docs
+        }, parentToChild -> {
+            assertEquals(0, parentToChild.getDocCount());
+            assertEquals(Double.POSITIVE_INFINITY, ((InternalMin) parentToChild.getAggregations().get("in_child")).getValue(),
+                    Double.MIN_VALUE);
+        });
+    }
+
+    public void testParentChild() throws IOException {
+        testCase(new MatchAllDocsQuery(), ParentToChildrenAggregatorTests::setupIndex, child -> {
+            assertEquals(4, child.getDocCount());
+            assertEquals(2, ((InternalMin) child.getAggregations().get("in_child")).getValue(), Double.MIN_VALUE);
+        });
+
+        testCase(new TermInSetQuery(UidFieldMapper.NAME, new BytesRef(Uid.createUid(PARENT_TYPE, "parent1"))),
+                ParentToChildrenAggregatorTests::setupIndex, child -> {
+                    assertEquals(1, child.getDocCount());
+                    assertEquals(10, ((InternalMin) child.getAggregations().get("in_child")).getValue(), Double.MIN_VALUE);
+                });
+
+        testCase(new TermInSetQuery(UidFieldMapper.NAME, new BytesRef(Uid.createUid(PARENT_TYPE, "parent2"))),
+                ParentToChildrenAggregatorTests::setupIndex, child -> {
+                    assertEquals(3, child.getDocCount());
+                    assertEquals(2, ((InternalMin) child.getAggregations().get("in_child")).getValue(), Double.MIN_VALUE);
+                });
+    }
+
+    private static void setupIndex(RandomIndexWriter iw) throws IOException {
+        iw.addDocument(createParentDocument("parent1"));
+        iw.addDocument(createChildDocument("child1", "parent1", 10));
+
+        iw.addDocument(createParentDocument("parent2"));
+        iw.addDocument(createChildDocument("child2", "parent2", 5));
+        iw.addDocument(createChildDocument("child3", "parent2", 2));
+        iw.addDocument(createChildDocument("child4", "parent2", 7));
```

Comment:
Sure, I will try to randomize this (number or parents, children per parent) in a simple way.